cobs.c

```diff
@@ -85,3 +85,37 @@ size_t cobs_decode(const uint8_t * restrict input, size_t length, uint8_t * rest
 
     return write_index;
 }
+
+/* Unstuffs "length" bytes of data at the location pointed to by
+ * "data", in-place, over-writing the original.
+ * Returns the number of bytes of the decoded data if it was
+ * successfully unstuffed, and 0 if there was an error unstuffing.
+ *
+ * Remove the "restrict" qualifiers if compiling with a
+ * pre-C99 C dialect.
+ */
+size_t cobs_decode_inplace(uint8_t * restrict data, size_t max_length) {
```

Comment:
`restrict` only makes sense if you have more than one pointer in the prototype. you can safely drop that.